spec/requests/articles_request_spec.rb

```diff
@@ -0,0 +1,136 @@
+    let(:article) { create(:article) }
+
+    it 'returns http success and contains one created article in json response' do
+      get article_path(article)
+
+      expect(response).to have_http_status(:ok)
+      expect(parsed_response['title']).to eq(article.title)
+      expect(parsed_response['body']).to eq(article.body)
+      expect(parsed_response['status']).to eq(article.status)
+    end
+  end
+
+  describe 'POST /create' do
+    context 'verify article creation' do
+      it 'returns http created and creates a new article when valid attributes passed' do
+        post articles_path, params: { article: valid_article_attributes }
+
+        expect(response).to have_http_status(:created)
+
+        expect(parsed_response['title']).to eq(valid_article_attributes[:title])
+        expect(parsed_response['body']).to eq(valid_article_attributes[:body])
+        expect(parsed_response['status']).to eq(valid_article_attributes[:status])
+      end
+
+      it 'returns http unprocessable entity and gives errors when invalid attributes passed' do
+        post articles_path, params: { article: invalid_article_attributes }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+
+        expect(parsed_response['body']).to include(/too short/)
+        expect(parsed_response['status']).to include(/is not included in the list/)
+      end
+    end
+  end
+
+  describe 'PUT /update' do
+    context 'verify article updation' do
+      let(:article) { create(:article) }
+
+      it 'returns http success and returns updated article when valid attributes passed' do
+        put article_path(article), params: { article: valid_article_attributes }
+
+        expect(response).to have_http_status(:ok)
+
+        expect(parsed_response['title']).to eq(valid_article_attributes[:title])
+        expect(parsed_response['body']).to eq(valid_article_attributes[:body])
+        expect(parsed_response['status']).to eq(valid_article_attributes[:status])
+      end
+
+      it 'returns http unprocessable entity and gives errors when invalid attributes passed' do
+        put article_path(article), params: { article: invalid_article_attributes }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+
+        expect(parsed_response['body']).to include(/too short/)
+        expect(parsed_response['status']).to include(/is not included in the list/)
+      end
+    end
+  end
+
+  describe 'DELETE /destroy' do
+    context 'verify article deletion' do
+      let(:article) { create(:article) }
+
+      it 'returns http no_content and deletes article' do
+        delete article_path(article)
+
+        expect(response).to have_http_status(:no_content)
+      end
+    end
+  end
+
+  describe 'GET /index?tag_id=<tag_id>' do
+    let(:tag)          { create(:tag, title: 'Common Tag Title') }
+    let(:titles)       { %w[First Second Third] }
+
+    before do
+      titles.each do |t|
+        create(:article, title: "#{t} title")
```

Comment:
You can also assign tags in the above line, rather than a second step.
```suggestion
        create(:article, title: "#{t} title", tags: [tag])
```

Reply:
This didn't work. Got the following error:
`     Failure/Error: create(:article, title: "#{t} title", tags: [tag])
     
     ActiveRecord::RecordInvalid:
       Validation failed: Taggings is invalid`